pkg/storage/chunk/client/local/fs_object_client.go

```diff
@@ -66,7 +66,8 @@ func NewFSObjectClient(cfg FSConfig) (*FSObjectClient, error) {
 func (FSObjectClient) Stop() {}
 
 func (f *FSObjectClient) ObjectExists(_ context.Context, objectKey string) (bool, error) {
-	_, err := os.Lstat(objectKey)
+	fullPath := filepath.Join(f.cfg.Directory, filepath.FromSlash(objectKey))
```

Comment:
I think our object key can contain a slash in it, e.g. fake/23432432:4324324324:432432432

For the file store we also base64 encode stuff.

Is all this accounted for here? (I'm also not sure what we do with this method itself in Loki and how likely this was already being used?)

Reply:
As discussed offline, this code now uses the same logic as the `GetObject` and `PutObject` methods.

This method isn't used a lot: 4 times in the Loki codebase. One is for a tool. The other usages is as part of an interface so it's probably used with a different backend there.